docs(partitioning): enhance best practices and time part templates do… (#5734)
influxdata/docs-v2: content/influxdb/cloud-dedicated/admin/custom-partitions/_index.md

---

When writing data to {{< product-name >}}, the InfluxDB v3 storage engine stores data in [Apache Parquet](https://parquet.apache.org/) format in the [Object store](/influxdb/cloud-dedicated/reference/internals/storage-engine/#object-store). Each Parquet file represents a _partition_--a logical grouping of data.
By default, InfluxDB partitions each table _by day_.
If this default strategy yields unsatisfactory performance for single-series queries,
you can define a custom partitioning strategy by specifying tag values and different time intervals to optimize query performance for your specific schema and workload.

- [Advantages](#advantages)
- [Disadvantages](#disadvantages)
- [Limitations](#limitations)
- [Plan for custom partitioning](#plan-for-custom-partitioning)
- [How partitioning works](#how-partitioning-works)
  - [Partition templates](#partition-templates)
  - [Partition keys](#partition-keys)
- [Partitions in the query life cycle](#partitions-in-the-query-life-cycle)
- [Partition guides](#partition-guides)
  {{< children type="anchored-list" >}}

> [!Note]
>
> #### When to consider custom partitioning
>

> > Consider custom partitioning if you have [optimized your queries](/influxdb/cloud-dedicated/query-data/troubleshoot-and-optimize/optimize-queries/) and single-series query (a query that returns a single measurement, tag set, and field key) performance is still unsatisfactory. 

Reviewer: The "series" definition in v3 is still a little ambiguous to me, but it is different than previous versions. With IOx, this really comes down to the primary key (timestamp and tag set), but timestamp is not included in the series definition. I'm wondering if this may be more accurate:

```suggestion
> Consider custom partitioning if you have [optimized your queries](/influxdb/cloud-dedicated/query-data/troubleshoot-and-optimize/optimize-queries/) and single-series query (a query that returns data with a specific tag value or tag set) performance is still unsatisfactory. 
```

Author: Thanks, I found the definition in the glossary, but couldn't remember if it was correct. I'll check and fix it.

  as it needs to compact more partition Parquet files.
- **Increased costs associated with [Object storage](/influxdb/cloud-dedicated/reference/internals/storage-engine/#object-storage)**
  as more partition Parquet files are created and stored.
- **Increased latency**, although slight, is linear with the total partition count in a table.
- **Risk of decreased performance for queries that don't use tags in the WHERE clause**.
  These queries might read many partitions and smaller files, which can degrade performance.

## Limitations

Custom partitioning has the following limitations:

- Define database and table partitions only during creation; you can't update the partition strategy afterward.
- Include a time part in a partition template.
- You can partition by up to eight dimensions (seven tags and a time interval).

## Plan for custom partitioning

After you have considered the [advantages](#advantages), [disadvantages](#disadvantages), and [limitations](#limitations) of
custom partitioning, use the guides in this section to:

1. Learn [how partitioning works](#how-partitioning-works)

> 2. Follow [best practices](/influxdb/cloud-dedicated/admin/custom-partitions/best-practices/),

Reviewer: ```suggestion
2. Follow [best practices](/influxdb/cloud-dedicated/admin/custom-partitions/best-practices/)
```
